src/daemon/utils_cache.c

```diff
@@ -913,6 +913,20 @@ int uc_iterator_get_interval(uc_iter_t *iter, cdtime_t *ret_interval) {
   return 0;
 } /* int uc_iterator_get_name */
 
+int uc_iterator_get_meta(uc_iter_t *iter, meta_data_t **ret_meta) {
+  if ((iter == NULL) || (iter->entry == NULL) || (ret_meta == NULL))
+    return -1;
+
+  if (iter->entry->meta == NULL) {
```

Comment:
`meta_data_clone()` gracefully deals with `NULL` arguments (by returning `NULL`), so this special case is not needed.